Introductory Tests/udp_initial_condition.py

```diff
@@ -145,6 +145,23 @@ def euler_approx(self, r, v, time_list, r_store, v_store, a_store):
         return r_store, v_store, a_store
 
 
+    def comet_collision_detection(self, x):
```

Comment:
the function is called collision_detection but returns true if there is no collision. That is confusing. I would call "point_is_outside_mesh" or something clear where it is clear that true means outside (or vice versa)